### PR TITLE
OpenCL implementation of IsEqual() function

### LINKED ISSUE #451: OpenCL implementation of IsEqual() function
We have CPU implementation of **IsEqual**() function in `src/image_function.h` and `src/image_function.cpp` files. We want to add OpenCL implementation of the same function. Please put new code in `src/opencl/image_function_opencl.h` and `src/opencl/image_function_opencl.cpp` files. Things which we need to do:
1. Copy 2 function declarations from `src/image_function.h` to `src/opencl/image_function_opencl.h`
2. Copy first function implementation from `src/image_function.cpp` to `src/opencl/image_function_opencl.cpp`
3. Add OpenCL kernel called `isEqualOpenCL` into unnamed namespace at the top of `src/opencl/image_function_opencl.cpp` file
4. Add second function implementation by following **Invert()** code logic

### PR DESCRIPTION
**Pull request by manekenpix**
Closes #451 

Added `IsEqual` to `src/opencl/image_function_opencl.h` and `src/opencl/image_function_opencl.cpp`.

Also, added OpenCL kernel `isEqualOpenCL` to `src/opencl/image_function_opencl.cpp`.

### CODE REVIEW

File: src/opencl/image_function_opencl.cpp [hunk 1]
```diff
@@ -882,6 +894,39 @@ namespace Image_Function_OpenCL
...
+        const uint8_t colorCount = Image_Function::CommonColorCount( in, out );
+        width = width * colorCount;
+
+        const uint32_t rowSizeIn  = in.rowSize();
+        const uint32_t rowSizeOut = out.rowSize();
+
+        const uint32_t offsetIn  = startYIn  * rowSizeIn  + startXIn  * colorCount;
+        const uint32_t offsetOut = startYOut * rowSizeOut + startXOut * colorCount;
+
+        uint32_t equal = 0;
```
Comment (ihhub) [inline src/opencl/image_function_opencl.cpp]: You cannot use CPU allocated variable for GPU kernels. Please use `multiCL::Type< uint32_t > differenceCount( 0u );` then pass into kernel as `differenceCount.data()` and compare it as `differenceCount.get() == 0`.

File: src/opencl/image_function_opencl.cpp [hunk 2]
```diff
@@ -882,6 +894,39 @@ namespace Image_Function_OpenCL
...
 
+    bool IsEqual( const Image & in1, const Image & in2 )
+    {
+        Image_Function::ParameterValidation( in1, in2 );
+
+        return Image_Function_Helper::IsEqual( IsEqual, in1, in2 );
+    }
+
+    bool IsEqual( const Image & in, uint32_t startXIn, uint32_t startYIn, const Image & out, uint32_t startXOut, uint32_t startYOut,
+                 uint32_t width, uint32_t height )
```
Comment (ihhub) [inline src/opencl/image_function_opencl.cpp]: Shift this line by 1 space to right.

**Comment by manekenpix** [comment]
@ihhub let me know if there's anything else I can do so the code can be merged.

**Comment by ihhub** [comment]
@manekenpix thank you very much for contribution to the project!

**Pull request merged**